lib/cache/cache.js

```diff
@@ -112,14 +112,12 @@ class Cache {
     // 5.5.2
     for (const response of responses) {
       // 5.5.2.1
-      const responseObject = new Response(response.body?.source ?? null)
-      const body = responseObject[kState].body
+      const responseObject = new Response(null)
       responseObject[kState] = response
-      responseObject[kState].body = body
       responseObject[kHeaders][kHeadersList] = response.headersList
       responseObject[kHeaders][kGuard] = 'immutable'
 
-      responseList.push(responseObject)
+      responseList.push(responseObject.clone())
```

Comment:
this is weird: this whole thing i'm kinda iffy about, somehow this was setting a 0-length uint8array as the source and was locking the stream.

Reply:
the cloning itself makes sense, if you matched a Response and then consumed its body, the cached stream would be locked if you matched it again.

Reply:
How is the body assigned to the new response?

Reply:
```js
      responseObject[kState] = response
```

`response` here is actually an internal response, not a Response object. It contains the body.